packages/edit-post/src/components/sidebar/post-schedule/index.js

```diff
@@ -3,18 +3,22 @@
  */
 import { __ } from '@wordpress/i18n';
 import { PanelRow, Dropdown, Button } from '@wordpress/components';
+import { useRef } from '@wordpress/element';
 import {
 	PostSchedule as PostScheduleForm,
 	PostScheduleLabel,
 	PostScheduleCheck,
 } from '@wordpress/editor';
 
 export function PostSchedule() {
+	const anchorRef = useRef();
+
 	return (
 		<PostScheduleCheck>
-			<PanelRow className="edit-post-post-schedule">
+			<PanelRow className="edit-post-post-schedule" ref={ anchorRef }>
 				<span>{ __( 'Publish' ) }</span>
 				<Dropdown
+					popoverProps={ { anchorRef: anchorRef.current } }
```

Comment:
Can you explain more why the default behavior of the Dropdown was not working as intended. (The toggle button being the anchorRef)?

Reply:
Ok I see, it's the width of the button. An alternative and maybe smaller solution would have been to right (or left never know) position the dropdown.

Reply:
The default `anchorRef` [is the button wrapper](https://github.com/WordPress/gutenberg/blob/c761c4c8bc70c29f55b781fa6c9d82e893dd6586/packages/components/src/dropdown/index.js#L87), `components-dropdown`. It's in a flex layout.

> An alternative and maybe smaller solution would have been to right (or left never know) position the dropdown.

I've tried passing different `position` props to `Dropdown` but that didn't seem to change the behavior. 😞 Or do you mean with CSS?

Reply:
yeah, well... setting the position at `left` or `right` doesn't work. First of all, I'm not sure whether it's possible to anchor the popover at the right because it's too close to the client border, so I guess it's going to apply for another `x` position automatically. 
I think we could set a fixed width at the button, maybe play a little bit with the text and element alignment, but it will change the button size which isn't something that we'd like to get either.

Reply:
Ok, than the solution in the PR looks good to me.